src/components/PathwayCourse/ResidentialCourse/index.js

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import useMediaQuery from "@mui/material/useMediaQuery";
+// import { breakpoints } from '../../theme/constant'
+import { actions as pathwayActions } from "../../PathwayCourse/redux/action";
+import { useDispatch } from "react-redux";
+import { breakpoints } from "../../../theme/constant";
+import {
+  Container,
+  Box,
+  Grid,
+  Card,
+  CardContent,
+  Typography,
+  Button,
+  Stack,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+import { interpolatePath, PATHS } from "../../../constant";
+import useStyles from "../styles";
+
+function ResidentialProgramme() {
+  const dispatch = useDispatch();
+  const { data } = useSelector((state) => state.Pathways);
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+  const classes = useStyles();
+
+  useEffect(() => {
+    dispatch(pathwayActions.getPathways());
+  }, [dispatch]);
+
+  let pathwayCourse;
+  data &&
+    data.pathways &&
+    data.pathways.forEach((pathway) => {
+      if (pathway.code === "PRCRSE") {
+        pathwayCourse = pathway.courses;
+      }
+    });
```

Comment:
In addition to the below suggestion, "PRCRSE" should probably be imported as a constant.
```suggestion
const resPathway = data && data.pathways && data.pathways.find((pathway) => pathway.code === "PRCRSE");
const pathwayCourse = resPathway && resPathway.courses;
```